tests/puzzle/test_solver.py

```diff
@@ -2165,7 +2183,8 @@ def test_solver_does_not_raise_conflict_for_locked_conditional_dependencies(
     repo.add_package(package_a)
     repo.add_package(package_b)
 
-    solver._locked = Repository("locked", [package_a])
+    dep_package_a = DependencyPackage(dependency_a, package_a)
```

Comment:
the changes here are maybe the only interesting point.  I've no idea what this testcase was trying to prove, or whether it is succeeding in proving it.  I've just replaced definiteliy-broken code with not-definitely broken code: but since the testcase passes with or without this I suspect it doesn't mean much.

On the plus side this is perhaps an argument in favour of typechecking the unit tests.  Presumably this test did something useful at some point, and it has been allowed to degrade because no-one noticed that `solver._locked` simply isn't a thing.  That wouldn't have happened, if the tests were being checked.